src/auth_proxy.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Observer module for Jenkins to auth_proxy integration."""
+
+import logging
+from typing import List
+
+import ops
+from charms.oathkeeper.v0.auth_proxy import AuthProxyConfig, AuthProxyRequirer
+from charms.traefik_k8s.v2.ingress import IngressPerAppRequirer
+
+import jenkins
+import state
+
+AUTH_PROXY_ALLOWED_ENDPOINTS: List[str] = []
+AUTH_PROXY_HEADERS = ["X-User"]
+
+
+logger = logging.getLogger(__name__)
+
+
+class Observer(ops.Object):
+    """The Jenkins Auth Proxy integration observer."""
+
+    def __init__(self, charm: ops.CharmBase, ingress: IngressPerAppRequirer):
+        """Initialize the observer and register event handlers.
+
+        Args:
+            charm: the parent charm to attach the observer to.
+            ingress: the ingress object from which to extract the necessary settings.
+        """
+        super().__init__(charm, "auth-proxy-observer")
+        self.charm = charm
+        self.ingress = ingress
+
+        self.auth_proxy = AuthProxyRequirer(self.charm)
+
+        self.charm.framework.observe(
+            self.charm.on["auth-proxy"].relation_joined, self._auth_proxy_relation_joined
```

Comment:
What happens for relation departed or changed?

Reply:
Jenkins won't be receiving any data though this relation, it will just write it, so no need to react on relation changed or departed events

Reply:
Hm...So if you remove the relation after Jenkins is configured in install_auth_proxy_config, there is no need to revert anything right?

Reply:
Mmm, you're right, we should revert the jenkins config change

Reply:
Updated